### DISCUSSION TITLE
How to make enum type of pgsql model orderable?

### DISCUSSION THREAD

**Discussion by Dennis-Zhang-SH** [prisma/prisma#18467] [Q&A]
say we have this kind of model:
```prisma
enum PrivilegeType {
    NONE
    VIEW
    EDIT
}
```
how to make it orderable? like EDIT > VIEW > NONE?

**Comment by nurul3101** [comment]
Hello @Dennis-Zhang-SH 👋 

Can you please elaborate on what you mean by making the enums Orderable?

**Reply by Dennis-Zhang-SH** [reply]
> Hello @Dennis-Zhang-SH 👋
> 
> Can you please elaborate on what you mean by making the enums Orderable?

Hi, what I mean is `enumsortorder`, and just the like example model, I can assign `enumsortorder` or it should be `EDIT > VIEW > NONE`.

**Comment by ludralph** [comment]
Hi @Dennis-Zhang-SH 👋 

One thing you could do is to create a model called `Privilege`. We can add a field to the database schema that maps each PrivilegeType value to a numerical value, and then use that field for ordering. For example, you could define the schema like this: 
```js
model Privilege {
  type         PrivilegeType
  type_order   Int
}
```
Based on this, you can create a record
```js
const newPrivilege = await prisma.privilege.create({
  data: {
    type: 'EDIT',
    type_order: 2
  }
})
```
Then you can use it for ordering your Prisma queries
```js
const privileges = await prisma.privilege.findMany({
  orderBy: {
    type_order: 'asc'
  }
})
```
This will order the Privilege records based on their type_order values, which correspond to the ordering of the PrivilegeType values.

Here is an example of what the output will look like
```js
[
  {
    id: 1,
    type: 'NONE',
    type_order: 0
  },
  {
    id: 2,
    type: 'VIEW',
    type_order: 1
  },
  {
    id: 3,
    type: 'EDIT',
    type_order: 2
  }
]
```

**Reply by Dennis-Zhang-SH** [reply]
> Hi @Dennis-Zhang-SH 👋
> 
> One thing you could do is to create a model called `Privilege`. We can add a field to the database schema that maps each PrivilegeType value to a numerical value, and then use that field for ordering. For example, you could define the schema like this:
> 
> ```js
> model Privilege {
>   type         PrivilegeType
>   type_order   Int
> }
> ```
> 
> Based on this, you can create a record
> 
> ```js
> const newPrivilege = await prisma.privilege.create({
>   data: {
>     type: 'EDIT',
>     type_order: 2
>   }
> })
> ```
> 
> Then you can use it for ordering your Prisma queries
> 
> ```js
> const privileges = await prisma.privilege.findMany({
>   orderBy: {
>     type_order: 'asc'
>   }
> })
> ```
> 
> This will order the Privilege records based on their type_order values, which correspond to the ordering of the PrivilegeType values.

Hi @ludralph, sure this can be done to order, but postgresql already had one `enumsortorder`, I wonder if prisma implemented this feature.

**Reply by ludralph** [reply]
There is an open feature request for [ordering by enums](https://github.com/prisma/prisma/issues/16968). `enumsortorder` isn't available in Prisma yet.
